src/internal/ppsutil/util.go

```diff
@@ -333,14 +333,14 @@ func FinishJob(pachClient *client.APIClient, jobInfo *pps.JobInfo, state pps.Job
 	// store commit states.
 	_, err := pachClient.RunBatchInTransaction(func(builder *client.TransactionBuilder) error {
 		if _, err := builder.PfsAPIClient.FinishCommit(pachClient.Ctx(), &pfs.FinishCommitRequest{
-			Commit: jobInfo.OutputCommit,
```

Comment:
Why does the MetaCommit have to come first?

Reply:
The reason for that change is mentioned in the main comment of the PR. I don't think this change is going to address the issue anymore with the goroutine per repo model though. I will probably need to add a wait on the meta commit in the code that handles the running / standby transition.

Reply:
So we start scaling pipelines before the Transaction is complete.  What if it's rolled back?

Reply:
No, the order of operations in a transaction correspond to the order the relevant events are sent to listeners when the transaction completes (operation order corresponds to the ordering of the row processing of the trigger which performs the notify calls on the relevant channels).